api/v1alpha1/clusterctl_config_types.go

```diff
@@ -36,6 +36,10 @@ type ClusterctlConfigSpec struct {
 	// Provider overrides
 	// +optional
 	Providers ProviderList `json:"providers,omitempty"`
+
+	// UseRancherDefaultRegistry indicates whether to use the Rancher default registry for pulling images.
+	// +optional
+	UseRancherDefaultRegistry bool `json:"useRancherDefaultRegistry,omitempty"`
```

Comment:
Should `systemDefaultRegistry` be used by default always? We could even skip including it in the spec and just read the setting directly in the controller code. From the setting description:

```
Private registry to be used for all Rancher System Container Images. If no value is specified, the default registry for the container runtime is used. For Docker and containerd, the default is `docker.io`.
```

What I think we can do is:
- If the setting value is empty, let Turtles use whatever registry is currently specified in the clusterctl config for community/prime provider images.
- If the setting value is not empty, Turtles should use it to substitute the image name.